src/index.js

```diff
@@ -1,610 +1,686 @@
-        window.rudderAnalyticsAddScript("https://polyfill-fastly.io/v3/polyfill.min.js?version=3.111.0&features=Symbol%2CPromise&callback=rudderAnalyticsMount");
-      } else {
-        window.rudderAnalyticsMount();
-      }
-      var loadOptions = {};
-      rudderanalytics.load(WRITE_KEY, DATAPLANE_URL, loadOptions);
+            var loadOptions = {};
+            rudderanalytics.load(WRITE_KEY, DATAPLANE_URL, loadOptions);
+        }
     }
-  }
 })();
 
 analytics.subscribe("product_viewed", (event) => {
-  const trackProperties = {
-    ...mapObjectKeys(event.data, productViewedEventMapping),
-  };
-  const contextualPayload = mapObjectKeys(
-    event.context,
-    contextualFieldMapping,
-  );
-
-  rudderanalytics.track("Product Viewed", trackProperties, contextualPayload);
+    const trackProperties = mapObjectKeys(
+        event.data,
+        productViewedEventMapping,
+    );
+    const contextualPayload = mapContextLineObjectKeys(
+        event.context,
+        contextualFieldMapping,
+    );
+    const shopifyDetailsObject = event?.data;
+
+    contextualPayload.shopifyDetails = shopifyDetailsObject;
+
+    rudderanalytics.track("Product Viewed", trackProperties, contextualPayload);
 });
 
 analytics.subscribe("cart_viewed", (event) => {
-  const lines = event?.data?.cart?.lines;
-  if (!lines) {
-    return;
-  }
-  const products = [];
-
-  lines.forEach((line) => {
-    const product = mapObjectKeys(line, cartViewedEventMapping);
-    products.push(product);
-  });
-
-  const trackProperties = {
-    products: products,
-    cart_id: event.data.cart.id,
-  };
-  const contextualPayload = mapObjectKeys(
-    event.context,
-    contextualFieldMapping,
-  );
-
-  rudderanalytics.track("Cart Viewed", trackProperties, contextualPayload);
+    const lines = event?.data?.cart?.lines;
+    if (!lines) {
+        return;
+    }
+    const products = [];
+
+    lines.forEach((line) => {
+        const trackProperties = mapObjectKeys(
+            line,
+            cartViewedEventMapping,
+        );
+        products.push(trackProperties);
+    });
+
+    const trackProperties = {
+        products: products,
+        cart_id: event?.data?.cart?.id,
+    };
+    const contextualPayload = mapContextLineObjectKeys(
+        event.context,
+        contextualFieldMapping,
+    );
+    const shopifyDetailsObject = event?.data;
+    contextualPayload.shopifyDetails = shopifyDetailsObject;
+
+    rudderanalytics.track("Cart Viewed", trackProperties, contextualPayload);
 });
 
 analytics.subscribe("product_added_to_cart", (event) => {
-  const trackProperties = {
-    ...mapObjectKeys(event.data, productToCartEventMapping),
-  };
-
-  const contextualPayload = mapObjectKeys(
-    event.context,
-    contextualFieldMapping,
-  );
+    const trackProperties = mapObjectKeys(
+        event.data,
+        productToCartEventMapping,
+    );
+    const contextualPayload = mapContextLineObjectKeys(
```

Comment:
For `event.context`, what's the use-case to maintain raw value & transformed value ?

Reply:
AFAIR, it was to give the added functionality that some shopify data point was either not mapped by us, or missed, then all of it will be there in the raw `data`. So if we can store whole of `data` object of an event in context, such fields can be located and used.
Another point was around cleanup of the properties, as with both RS mapped properties and actual shopify properties, there can be some confusion and difficulty to go through them due to the large number.

Reply:
No duplicate keys - transformed properties along with untransformed properties are being maintained